compat/src/react-18-hooks.js

```diff
@@ -0,0 +1,54 @@
+import { useState, useLayoutEffect, useEffect } from 'preact/hooks';
+
+/**
+ * Asynchronously schedule a callback
+ * @type {(cb: () => void) => void}
+ */
+/* istanbul ignore next */
+// Note the following line isn't tree-shaken by rollup cuz of rollup/rollup#2566
+const defer =
+	typeof Promise == 'function'
+		? Promise.prototype.then.bind(Promise.resolve())
+		: setTimeout;
+
+export function useDeferredValue(val) {
+	return val;
+}
+
+export function useTransition() {
+	const [isPending, setIsPending] = useState(false);
+
+	const start = cb => {
+		setIsPending(true);
+		cb();
+		defer(() => {
```

Comment:
I think this will behave very differently from React.
In React, all state updates inside the callback will be deferred and `isPending` signals that the current state is not up to date.

If the idea is just to stub, the hook without providing the functionality, it might be better to simply do something like:
```js
```export function useTransition() {
  return [false, useCallback(cb => { cb(); }, [])];
}
```

Reply:
In particular, it doesn’t make sense to set isPending to true and immediately perform the state update. The isPending flag being true is supposed to mean the state update has _not_ been applied.

Reply:
I recommend to take some existing useTransition demo and test the behavior. For example https://github.com/reactwg/react-18/discussions/65.

I think a best-effort attempt could be to do let it paint once with isPending = true and old state, and then after paint, let it update to the final state. This will have some downsides though:

1. The transition update itself would still be non-interruptible. So even if it’s delayed, you’re still going to lock the screen right after. So it doesn’t really improve responsiveness except maybe communicating there’ll be a slowdown right after.

2. Delaying the callback itself seems like significantly different behavior. React doesn’t do that.

3. This still doesn’t handle the (arguably) more important use case in longer term — preventing Suspense fallback from showing. startTransition is an important part of implementing common UI patterns with Suspense.

So all in all, maybe providing an empty naïve shim would be better than doing something half-baked.